### PR TITLE
Device classes, entity names and more complete equalizer support

### PR DESCRIPTION
**Pull request by olalid**
Entities now have device_class set where it makes sense.
Entity names are not on the format of "<charger/equalizer name> <sensor name>".
Equalizer support reworked with several separate sensors and better integration in the controller.
Cleaner implementation of energy monitors scheduling.

### CODE REVIEW

**Comment by astrandb** [comment]
I have no consumption sensors defined and get this:
```
 File "/home/ake/.homeassistant/custom_components/easee/controller.py", line 225, in refresh_consumption_sensors
    self.consumption_sensor_entities[self.next_consumption_sensor].async_schedule_update_ha_state(True)
IndexError: list index out of range
```

File: custom_components/easee/const.py [hunk 1]
```diff
@@ -119,14 +132,16 @@
         ],
         "units": None,
         "convert_units_func": None,
+        "device_class": DEVICE_CLASS_SIGNAL_STRENGTH,
         "icon": "mdi:wifi",
```
Comment (astrandb) [inline custom_components/easee/const.py:136]: I suggest DEVICE_CLASS_CONNECTIVITY - import from homeassistant.components.binary_sensor. You can test with Icon: None also

**Comment by olalid** [comment]
> I have no consumption sensors defined and get this:
> 
> ```
>  File "/home/ake/.homeassistant/custom_components/easee/controller.py", line 225, in refresh_consumption_sensors
>     self.consumption_sensor_entities[self.next_consumption_sensor].async_schedule_update_ha_state(True)
> IndexError: list index out of range
> ```

Added a check for empty array, so should be solved now.

File: custom_components/easee/const.py (hunk 1 above)
Comment (olalid) [reply custom_components/easee/const.py]: Yes, better. Changed now. Thanks.

File: custom_components/easee/const.py [hunk 2]
```diff
@@ -50,9 +63,10 @@
         ],
         "units": None,
         "convert_units_func": None,
+        "device_class": None,
```
Comment (astrandb) [inline custom_components/easee/const.py:66]: DEVICE_CLASS_LOCK

File: custom_components/easee/const.py [hunk 3]
```diff
@@ -326,6 +359,111 @@
...
+            "state.latestPulse",
+            "state.clockAndDateMeter",
+            "state.rcpi",
+            "state.localRSSI",
+            "state.softwareRelease",
+            "state.latestFirmware",
+        ],
+        "units": None,
+        "convert_units_func": None,
+        "device_class": DEVICE_CLASS_SIGNAL_STRENGTH,
```
Comment (astrandb) [inline custom_components/easee/const.py:378]: DEVICE_CLASS_CONNECTIVITY

**Review by astrandb — COMMENTED** [review]
A few more device_classes

File: custom_components/easee/const.py (hunk 3 above)
Comment (olalid) [reply custom_components/easee/const.py:378]: I do not think that is possible on a sensor, only binary_sensor?

Comment (astrandb) [reply custom_components/easee/const.py:378]: I missed that it was a sensor.

File: custom_components/easee/const.py [hunk 4]
```diff
@@ -326,6 +359,111 @@
...
+        "convert_units_func": "round_1_dec",
+        "device_class": DEVICE_CLASS_VOLTAGE,
+        "icon": None,
+        "state_func": lambda state: float(
+            max(
+                state["voltageNL1"]
+                if state["voltageNL1"] is not None
+                else 0.0,
+                state["voltageNL2"]
+                if state["voltageNL2"] is not None
```
Comment (fondberg) [inline custom_components/easee/const.py:417]: can be written
`state["voltageNL2"] or 0.0`

**Review by fondberg — APPROVED** [review]

**Review by fondberg — APPROVED** [review]

**Pull request merged**